packages/esm-patient-common-lib/src/offline/patient.ts

```diff
@@ -8,53 +8,66 @@ export const patientRegistrationSyncType = 'patient-registration';
  * in offline mode by the patient registration module.
  */
 export function usePatientOrOfflineRegisteredPatient(patientUuid: string): ReturnType<typeof usePatient> {
+  const isOnline = useConnectivity();
   const onlinePatientState = usePatient(patientUuid);
-  const offlinePatientState = useSWR(`offlineRegisteredPatient/${patientUuid}`, async () => {
-    const offlinePatient = await getOfflineRegisteredPatientAsFhirPatient(patientUuid);
-    if (!offlinePatient) {
-      throw new Error(`No offline registered patient could be found. UUID: ${patientUuid}`);
+  const [offlinePatientState, setOfflinePatientState] = useState<{ data: fhir.Patient | null; error: Error | null }>({
+    data: null,
+    error: null,
+  });
+
+  useEffect(() => {
+    if (isOnline) {
+      setOfflinePatientState({ error: null, data: null });
+    } else {
+      getOfflineRegisteredPatientAsFhirPatient(patientUuid)
+        .then((offlinePatient) => {
+          setOfflinePatientState({ error: null, data: offlinePatient });
+        })
+        .catch((err) => {
+          setOfflinePatientState({ error: err, data: null });
+        });
     }
+  }, [patientUuid, isOnline]);
 
-    return offlinePatient;
-  });
+  return useMemo(() => {
+    if (onlinePatientState.isLoading || (!offlinePatientState.data && !offlinePatientState.error)) {
+      return {
+        isLoading: true,
+        patient: null,
+        patientUuid,
+        error: null,
+      };
+    }
 
-  if (onlinePatientState.isLoading || (!offlinePatientState.data && !offlinePatientState.error)) {
-    return {
-      isLoading: true,
-      patient: null,
-      patientUuid,
-      error: null,
-    };
-  }
+    if (onlinePatientState.patient && !(onlinePatientState.patient as any).issue) {
+      return {
+        isLoading: false,
+        patient: onlinePatientState.patient,
+        patientUuid,
+        error: null,
+      };
+    }
 
-  if (onlinePatientState.patient && !(onlinePatientState.patient as any).issue) {
-    return {
-      isLoading: false,
-      patient: onlinePatientState.patient,
-      patientUuid,
-      error: null,
-    };
-  }
+    if (offlinePatientState.data) {
+      return {
+        isLoading: false,
+        patient: offlinePatientState.data,
```

Comment:
If `offlinePatientState.data` is supposed to contain multiple things I would change this to `offlinePatientState.data.patient`. Otherwise I would clean up as `offlinePatientState.patient`

Reply:
Yeah... let me fix that... this is an artifact left over from replacing `useSWR()` with an initially identical API so I didn't screw up the rest of the logic.